### PR TITLE
Introduce "stage3" vs. "experimental" distinction

### PR DESCRIPTION
**Pull request by tchetwin**
Following https://github.com/estree/estree/issues/240#issuecomment-778168995, this promotes the currently "experimental" class features AST specification in to the root.

### CODE REVIEW

**Comment by nzakas** [comment]
This isn’t the correct way to promote nodes. All files have an ES version number, so this should probably be es2022.md. Please see es2021.md as an example for how to set up the file.

**Review by JLHwung — COMMENTED** [review]
I think we should create es2022 only if class features reaches stage 4 officially after es2021 is standardized. Before that it is experimental by definition.

**Comment by tchetwin** [comment]
I'm somewhat inclined to agree, picking `es2021`/`es2022`/`es20**` is forecasting. Perhaps just `esnext.md`?

**Comment by RReverser** [comment]
@tchetwin That's essentially what individual features in experimental folder represent - any ESNext features that are not yet finalized in any ES version.

Another doc for the same purpose seems unnecessary.

**Comment by bradzacher** [comment]
The problem is that "experimental" is unclear. Does it mean the AST is experimental, the feature is experimental, or both? 

Some clarity should be added, esp if the intention is to allow stage 3 features to be considered stable (see #240). 

If a new file called "esnext" isn't desired, then maybe a separate folder called such is better? Something to mark an AST spec as "ready for use but feature is stage 3, not stage 4" so it's separate from the "still experimental and subject to change" ast specs.

**Comment by nzakas** [comment]
I do think it's useful to have a distinction between "experimental" (stage 0-2) and "baked" (stage 3-4). We know that stage 3 features are effectively done and so if we are happy with the AST format for those, it would be nice to have some way to flag those as separate from the < stage 3 features.

If we don't want an `esnext.md`, then perhaps a `stage3` folder could serve the same purpose?

**Comment by JLHwung** [comment]
> a stage3 folder could serve the same purpose?

That works for me. On those PRs moving from `experimental` to `stage3`, we should reach consensus on the AST shapes and it can be considered stable for implementers, with a condition that the AST will only change unless the spec has undergone significant changes from stage 3 to stage 4. By doing that we are effectively advancing AST shape consensus from stage 4 to stage 3 proposals.

**Comment by tchetwin** [comment]
I am *very* happy to see us draw a distinction between `experimental` (stages <3) and `stage3`. I think sending this signal will make it a lot easier for tooling to confidently make progress on stage3 features and themselves provide feedback in the stage advancement process.

**Comment by RReverser** [comment]
Separate `stage3` sounds reasonable to me. I wonder if it should be a single doc instead of a folder perhaps - since it's almost same as stable `es*.md` versions?

**Comment by tchetwin** [comment]
> then perhaps a `stage3` folder could serve the same purpose?

> I wonder if it should be a single doc instead of a folder perhaps

Could go either way, `stage3.md` or `stage3/class-features.md`

A single file might "force" reconciliation of proposals that touch similar parts of ASTs. Imagine if [class-static-block](https://github.com/estree/estree/blob/master/experimental/class-static-block.md) was also stage 3, it also changes `ClassBody`, having it in a different file might be confusing. This reconciliation would be similar to how the `acorn-stage3` collects various underlying plugins under one umbrella.

Counter to that: if one of two interlinked proposals makes it from stage3 -> stage4 before the other, they then need to be untangled again. Some proposals (like class-features) have spent a long time in stage3 and have been overtaken by other proposals.

File: stage3/class-features.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
+# Classes
```
Comment (JLHwung) [inline stage3/class-features.md:1]: Can you separate this one to another PR? So this PR focuses on creating the `stage3` thing and in another PR we seek consensus on the specific class features.

Comment (tchetwin) [reply stage3/class-features.md:1]: I can do, these aren't really material changes to the class features themselves though, just a massaging of the format to more closely resemble other files.

Comment (tchetwin) [reply stage3/class-features.md:1]: I guess you're implicitly advocating for the `stage3/*.md` model as otherwise there is no other change to be made.

**Comment by tchetwin** [comment]
On @JLHwung's recommendation I have removed the class-features specifics so this PR now only introduced the concept of a distinction between "stage3" and "experimental". Class features promotion will follow in another PR.

**Review by JLHwung — APPROVED** [review]

**Comment by nzakas** [comment]
I think it's a good idea to keep separate files in the `stage3` folder like we do in `experimental` because it will limit the number of time we need to copy-paste into a single file. :+1:

**Review by nzakas — APPROVED** [review]

**Review by RReverser — APPROVED** [review]
